src/collections/_documentation/platforms/php/index.md

```diff
@@ -0,0 +1,159 @@
+to create relative paths.
+
+## Transport
+
+Transports are the classes in Sentry PHP that are responsible for communicating
+with a service in order to deliver an event. There are several types of transports
+available out-of-the-box, all of which implement the `TransportInterface`
+interface;
+
+- The `NullTransport` which is used in case you do not define a `DSN` in the options. It will not send events.
+- The `HttpTransport` which is the default and will be used when the server
+  is set in the client configuration.
+- The `SpoolTransport` which can be used to defer the sending of events (e.g.
+  by putting them into a queue).
+  
+The examples below pretty much replace the `init()` call.
+
+### NullTransport
+
+Although not so common there could be cases in which you don't want to send
+events at all. The `NullTransport` transport does this: it simply ignores
+the events, but report them as sent.
+
+```php
+use Sentry\ClientBuilder;
+use Sentry\Transport\NullTransport;
+use Sentry\State\Hub;
+
+$transport = new NullTransport();
+$builder = ClientBuilder::create(['dsn' => '___PUBLIC_DSN___']);
+$builder->setTransport($transport);
+Hub::getCurrent()->bindClient($builder->getClient());
+```
+{% capture __note %}
+> Note: `Hub::getCurrent()->bindClient($builder->getClient());` is required to tell the global function calls to use your new `Client` instead.
+{% endcapture %}
+
+{{ __note }}
+
+### HttpTransport
+
+The `HttpTransport` sends events over the HTTP protocol using [Httplug](http://httplug.io/) the
+best adapter available is automatically selected when creating a client instance
+through the client builder, but you can override it using the appropriate methods.
+
+```php
+use Sentry\ClientBuilder;
+use Sentry\Transport\HttpTransport;
+use Sentry\State\Hub;
+
+$transport = new HttpTransport($configuration, HttpAsyncClientDiscovery::find(), MessageFactoryDiscovery::find());
+$builder = ClientBuilder::create(['dsn' => '___PUBLIC_DSN___']);
+$builder->setTransport($transport);
+Hub::getCurrent()->bindClient($builder->getClient());
+```
+
+{{ __note }}
+
+### SpoolTransport
+
+The default behavior is to send events immediately. You may, however, want to
+avoid waiting for the communication to the Sentry server that could be slow
+or unreliable. This can be avoided by choosing the `SpoolTransport` which
+stores the events in a queue so that another process can read it and and take
+care of sending them. Currently only spooling to memory is supported.
+
+
+```php
+use Sentry\ClientBuilder;
+use Sentry\Transport\SpoolTransport;
+use Sentry\State\Hub;
+
+$spool = new MemorySpool();
+$transport = new SpoolTransport($spool);
+$builder = ClientBuilder::create(['dsn' => '___PUBLIC_DSN___']);
+$builder->setTransport($transport);
+Hub::getCurrent()->bindClient($builder->getClient());
+
+// When the spool queue is flushed the events are sent using the transport
+// passed as parameter of the flushQueue method.
+  
+$spool->flushQueue($transport);
```

Comment:
This line is wrong and will create a circular dependency while sending the events because the `$transport` variable is referring to the `SpoolTransport`, so you're basically saying "put the event into a queue and then when it's flushed put the event in the same queue again". Instead you need two transports, one that is the `SpoolTransport` and one that will actually send for real the events or drop them (`HttpTransport`, `NullTransport` or a custom one)